wsf_scraping/spiders/gov_spider.py

```diff
@@ -10,20 +11,47 @@ class GovSpider(BaseSpider):
         'JOBDIR': 'crawls/gov_uk'
     }
 
+    def __init__(self, *args, **kwargs):
+        year_before = kwargs.get('year_before', False)
+        year_after = kwargs.get('year_after', False)
+        id = kwargs.get('uuid', '')
+
+        self.uuid = id
+        if year_before:
+            self.year_before = year_before
+        else:
+            self.year_before = ''
+
+        if year_after:
+            self.year_after = year_after
+        else:
+            self.year_after = ''
+
     def start_requests(self):
         """Sets up the base urls and start the initial requests."""
-        urls = [
-            'https://www.gov.uk/government/publications',
-        ]
-
-        for url in urls:
-            self.logger.info('Initial url: %s', url)
-            yield scrapy.Request(
-                url=url,
-                callback=self.parse,
-                errback=self.on_error,
-                dont_filter=True,
-            )
+        url = 'https://www.gov.uk/government/publications'
+
+        if self.year_before or self.year_after:
```

Comment:
double checking here: what happens if only one of `year_before` or `year_after` is assigned? should this be an `and` instead?

Reply:
No it's perfectly normal here. If only one of these is specified, it returns everything before x or everything after x

Reply:
👍